static-assets/scripts/main.js

```diff
@@ -533,8 +599,34 @@
         );
       };
 
+      this.editSiteData = function(site, onEditSuccess) {
+        const eventIdSuccess = 'editSiteDialogSuccess';
+        CrafterCMSNext.system.store.dispatch({
+          type: 'SHOW_EDIT_SITE_DIALOG',
+          payload: {
+            open: true,
+            site,
+            onSaveSuccess: {
+              type: 'BATCH_ACTIONS',
+              payload: [
+                {
+                  type: 'DISPATCH_DOM_EVENT',
+                  payload: { id: eventIdSuccess }
+                },
+                { type: 'CLOSE_EDIT_SITE_DIALOG' }
+              ]
+            }
+          }
+        });
+        CrafterCMSNext.createLegacyCallbackListener(eventIdSuccess, (response) => {
+          $rootScope.showNotification(formatMessage(adminDashboardMessages.siteUpdated));
+          onEditSuccess(response);
+        });
```

Comment:
These callbacks aren't cleaned up if the dialog is dismissed without success. We need to add an additional listener for the dismiss to clean the success handler.
```js
let cleanupSuccess;
cleanupSuccess = CrafterCMSNext.createLegacyCallbackListener(...);
CrafterCMSNext.createLegacyCallbackListener(...
  cleanupSuccess()
});
```
```